snackpub/src/index.ts

```diff
@@ -160,3 +136,89 @@ async function runAsync() {
     process.exit(1);
   }
 })();
+
+/**
+ * Register process signals to shutdown the server
+ */
+function registerShutdownHandlers(server: Server, redisClients: NullableRedisClientType[]) {
+  const shutdownSignals: NodeJS.Signals[] = [
+    'SIGHUP',
+    'SIGINT',
+    'SIGTERM',
+    'SIGUSR2', // Nodemon sends SIGUSR2 when it restarts the process it's monitoring
+  ];
+
+  for (const signal of shutdownSignals) {
+    process.once(signal, (signal: NodeJS.Signals) => {
+      console.log(
+        `Received ${signal}; the HTTP server is shutting down and draining existing connections`
+      );
+      shutdownAsync(server, redisClients);
+    });
+  }
+}
+
+/**
+ * Register Redis client handlers
+ */
+function registerRedisClientErrorHandlers(redisClients: NullableRedisClientType[]) {
+  for (const redisClient of redisClients) {
+    redisClient?.on('error', async (error) => {
```

Comment:
Looks like this doesn't need to be async?
```suggestion
    redisClient?.on('error', (error) => {
```